test/index.js

```diff
@@ -698,6 +698,213 @@ describe("mongoose_delete with options: { deletedId : true, deletedIdType: Strin
+      should.exist(notDeleted);
+      notDeleted.deleted.should.equal(false);
+    } catch (err) {
+      await session.abortTransaction();
+      should.not.exist(err);
+    } finally {
+      session.endSession();
+    }
+  });
+
+  it("deleteById() -> should support transactions with session parameter", async function () {
+    if (mongooseMajorVersion < 5) {
+      return this.skip();
+    }
+    
+    const session = await mongoose.startSession();
+    try {
+      session.startTransaction();
+      
+      const doc = await TestModel.findOne({ name: 'Yoda' });
+      
+      // Delete by ID with session
+      const result = await TestModel.deleteById(doc._id, { 
+        deletedBy: userId,
+        session: session
+      });
+      
+      // Verify operation was successful
+      expect(result).to.be.mongoose_ok();
+      expect(result).to.be.mongoose_count(1);
+      
+      // Commit the transaction
+      await session.commitTransaction();
+      
+      // Verify document was deleted after commit
+      const deleted = await TestModel.findOneDeleted({ name: 'Yoda' });
+      deleted.deleted.should.equal(true);
+    } catch (err) {
+      should.not.exist(err);
+    } finally {
+      session.endSession();
+    }
+  });
+
+  it("delete() with multiple documents -> should support transactions with session parameter", async function () {
+    if (mongooseMajorVersion < 5) {
+      return this.skip();
+    }
+    
+    const session = await mongoose.startSession();
+    try {
+      session.startTransaction();
+      
+      // Delete multiple documents with session
+      const result = await TestModel.delete(
+        { name: { $in: ['Anakin Skywalker', 'Obi-Wan Kenobi'] } }, 
+        { 
+          deletedBy: userId, 
+          session: session
+        }
+      );
+      
+      // Verify operation was successful
+      expect(result).to.be.mongoose_ok();
+      expect(result).to.be.mongoose_count(2);
+      
+      // Count deleted docs within the transaction
+      const countInTxn = await TestModel.countDocuments({ deleted: false },  { session });
+      countInTxn.should.equal(1); // Only Yoda remains
+      
+      // Abort transaction
+      await session.abortTransaction();
+      
+      // Verify no documents were deleted after rollback
+      const count = await TestModel.countDocuments();
+      count.should.equal(3);
+    } catch (err) {
+      await session.abortTransaction();
+      should.not.exist(err);
+    } finally {
+      session.endSession();
+    }
+  });
+
+  it("restore() -> should support transactions with session parameter", async function () {
+    if (mongooseMajorVersion < 5) {
+      return this.skip();
+    }
+    
+    // First delete a document
+    const doc = await TestModel.findOne({ name: 'Obi-Wan Kenobi' });
+    await doc.delete({ deletedBy: userId });
+    
+    const session = await mongoose.startSession();
+    try {
```

Comment:
🤏 I think the try should be higher here 

```suggestion
    }
    try {
    // First delete a document
    const doc = await TestModel.findOne({ name: 'Obi-Wan Kenobi' });
    await doc.delete({ deletedBy: userId });
    
    const session = await mongoose.startSession();
```

Reply:
I've moved the `findOne` and `delete` operations into the try but left the session out of the try clause because we need access to it in the finally

Reply:
oh yeah good point